org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/files/generateAdaptorServletListener.mtl

```diff
@@ -69,14 +71,23 @@ import [javaClassFullNameForAdaptorManager(anAdaptorInterface) /];
 // [protected ('imports')]
 // [/protected]
 
+/**
+ * During the initialization of this ServletListener, the base URI for the OSLC resources produced by this server is configured through the OSLC4J method setPublicURI().
+ * A number of approaches can be used to set this base URI:
+ * 1. Set a system environment property of name "LYO_BASE".
+ * 2. In the web.xml file, set a "context-param" property, with name "[javaClassPackageNameForAdaptorServletListener(anAdaptorInterface) /].baseurl"
+ * 3. Override the value of the DEFAULT_BASE constant.
+ */
 public class [javaClassNameForAdaptorServletListener(anAdaptorInterface) /] implements ServletContextListener  {
-    private static final String DEFAULT_BASE = "http://localhost:8080";
-    private static final String PROPERTY_BASE = servletContextParameterName("baseurl");
-    private static final Logger logger = Logger.getLogger([javaClassNameForAdaptorServletListener(anAdaptorInterface) /].class.getName());
 
-    //If you are using another servletName in your web.xml configuration file, modify this variable early in the method contextInitialized below
+    //These are default names. You can modify any of these "constants" early in the contextInitialized() method.
+    private static String BASE_PATH_ENV_KEY = "LYO_BASE";
+    private static String DEFAULT_BASE = "http://localhost:8080";
```

Comment:
I think this variable could then be renamed into `fallbackBase`

Reply:
Yes, but I can't be bothered. Might as well keep it for familiarity reasons. There are many methods and variables I ould rather rename.

Reply:
They need to be inlined anyway.

Reply:
Fixed